FLAnimatedImage/IndexedImageCache.h

```diff
@@ -0,0 +1,26 @@
+//
+//  IndexedImageCache.h
+//  FLAnimatedImage
+//
+//  Created by Sih Ou-Yang on 2024/2/23.
+//  Copyright © 2024 com.cardinalblue. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+@interface IndexedImageCache : NSObject
+
+- (instancetype _Nonnull)initWithLimit:(NSInteger)limit; // limit represents the size of the cache
+
+- (void)add:(UIImage *_Nonnull)image withIndex:(NSInteger)index;
+
+- (void)set:(UIImage *_Nullable)image atIndex:(NSInteger)index;
```

Comment:
Just keep set function?
```objc
- (void)set:(UIImage *_Nullable)image atIndex:(NSInteger)index;
```

Reply:
I guess `set` should be private and can't appear in the header?

Reply:
I guess `set` should be private and can't appear in the header?